v2/internal/reconcilers/azure_generic_arm_reconciler.go

```diff
@@ -80,6 +80,7 @@ type AzureDeploymentReconciler struct {
 	PositiveConditions *conditions.PositiveConditionBuilder
 	config             config.Values
```

Comment:
Ha, I guess this is me doing the exported/unexported thing on `AzureDeploymentReconciler` that I'm complaining about above, but I think passing the generic reconciler would avoid that?

Reply:
As mentioned above, can't pass it as it'd created a circular reference, and passing it goes against the actual intent of the relationship of these objects. The real solution is to construct `AzureDeploymentReconciler` first and pass _that_ to `GenericReconciler`. Then we can delete almost all of the `GenericReconciler` parameters as it doesn't need them directly (it only needs them to give them to `AzureDeploymentReconciler`.

Reply:
Also, all of these should probably be unexported. I've made a note to fix that but will do it in a separate PR